x/distribution/spec/05_hooks.md

```diff
@@ -5,26 +5,44 @@ order: 5
 # Hooks
 
 ## Create or modify delegation distribution
- 
+
  - triggered-by: `staking.MsgDelegate`, `staking.MsgBeginRedelegate`, `staking.MsgUndelegate`
 
-The pool of a new delegator bond will be 0 for the height at which the bond was
-added, or the withdrawal has taken place. This is achieved by setting
-`DelegationDistInfo.WithdrawalHeight` to the height of the triggering transaction. 
+### Before
+
+The delegation rewards are withdrawn to the delegator's withdraw address.
+The rewards include the current period, but excludes the starting period.
+Then increment the validator period, which is used for calculating rewards.
+The latter is necessary because, due to the nature of the triggers, the validator's power and share distribution can have changed.
+The reference count for the delegator's starting period is decremented.
+
+### After
+
+The starting height of the delegation is set to the previous period.
+Because of the `Before`-hook, this is the last period for which the delegator was rewarded.
+
+## Validator created
+
+- triggered-by: `staking.MsgCreateValidator`
+
+Initialized the validator variables: Historical rewards, current accumulated rewards, accumulated commission, and total outstanding rewards.
+All are set to a `0` by default.
+The period is set to `1`.
+
+## Validator removed
+
+- triggered-by: `staking.RemoveValidator`
 
-## Commission rate change
- 
- - triggered-by: `staking.MsgEditValidator`
+Any outstanding commission is sent to the validator's self-delegation's withdrawal address.
+Any remaining rewards still stored in the validator gets sent to the community fee pool.
 
-If a validator changes its commission rate, all commission on fees must be
-simultaneously withdrawn using the transaction `TxWithdrawValidator`.
-Additionally the change and associated height must be recorded in a
-`ValidatorUpdate` state record.
+Note: the validator only gets removed when it has no remaining delegations.
+All outstanding rewards have been withdrawn.
 
-## Change in Validator State
- 
- - triggered-by: `staking.Slash`, `staking.UpdateValidator`
+## Validator is slashed
 
-Whenever a validator is slashed or enters/leaves the validator group all of the
-validator entitled reward tokens must be simultaneously withdrawn from
-`Global.Pool` and added to `ValidatorDistInfo.Pool`. 
+- triggered-by: `staking.Slash`
+  
+The current validator period reference count is incremented, since the slash event will need to refer to it.
+The validator period is incremented.
+The slash event is stored, to be referenced when calculating delegator rewards.
```

Comment:
```suggestion
The current validator period reference count is incremented so the slash event reference to it is accurate.
The validator period is incremented.
The slash event is stored and is referenced when calculating delegator rewards.
```

Reply:
Thank you. I think some things can be made still clearer. Let me know what you think of my new changes.